aws/aws-how-to/instances/data-science-stack-on-ec2.rst

```diff
@@ -0,0 +1,144 @@
+
+.. code::
+
+    ssh -i <<YOUR_KEYPAIR>> ubuntu@<<YOUR_MACHINE_IP>>
+
+
+If you are connecting from Windows, you can use PuTTy.
+
+Once you have connected, run a full upgrade:
+
+.. code::
+
+    sudo apt update && sudo apt upgrade -y
+
+
+If you get a new kernel, it is advised to restart the machine before proceeding.
+
+Now install the GPU drivers:
+
+.. code::
+
+    sudo apt install -y ubuntu-drivers-common
+    sudo ubuntu-drivers install
+    sudo reboot
+
+
+After reboot, check if the drivers and CUDA have been installed properly and the GPU is detected correctly:
+
+.. code::
+
+    nvidia-smi
+
+
+The output should be similar to:
+
+.. code-block:: none
+
+        +---------------------------------------------------------------------------------------+
+        | NVIDIA-SMI 535.104.05             Driver Version: 535.104.05   CUDA Version: 12.2     |
+        |-----------------------------------------+----------------------+----------------------+
+        | GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
+        | Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
+        |                                         |                      |               MIG M. |
+        |=========================================+======================+======================|
+        |   0  Tesla T4                       On  | 00000000:00:1E.0 Off |                    0 |
+        | N/A   26C    P8               9W /  70W |      2MiB / 15360MiB |      0%      Default |
+        |                                         |                      |                  N/A |
+        +-----------------------------------------+----------------------+----------------------+
+
+        +---------------------------------------------------------------------------------------+
+        | Processes:                                                                            |
+        |  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
+        |        ID   ID                                                             Usage      |
+        |=======================================================================================|
+        |  No running processes found                                                           |
+        +---------------------------------------------------------------------------------------+
+
+Note that the GPU is properly detected and CUDA has also been installed.
+
+Install MicroK8s and DSS
+------------------------
+
+Install MicroK8s:
+
+
+.. code::
+
+    sudo snap install microk8s --channel 1.28/stable --classic
+    sudo microk8s enable storage dns rbac
+    sudo microk8s enable gpu
+
+Install DSS:
+
+.. code::
+
+    sudo snap install data-science-stack --channel latest/stable
+    dss initialize --kubeconfig="$(sudo microk8s config)"
```

Comment:
i would prefer to reference the official DSS docs for this

Reply:
TLDR; I'll leave that to @k-dimple 
In a conversation with another team (Anbox), we were discussing how referencing other guides sometimes impacted the experience, as a 1/the official guides tend to be more generic and sometimes confuse the customer with other things, and 2/ the How-to stops to be a straightforward recipe. 

The documentation person there told me that they are trying now to have self-sufficient guides for that reason.

Reply:
Yeah, I agree with the argument. If the whole how-to guide was just about installing DSS, we could have directly pointed to the DSS docs. Here since we are talking about specific EC2 instances and including specific drivers, it is better to give a complete set of instructions and to point them to the DSS docs for further reference. (Carlos has already done this by adding a link in the introduction section.) So I think we can keep it in its current state.